### DISCUSSION TITLE
SSH config location

### DISCUSSION THREAD

**Discussion by Glideh** [garethgeorge/backrest#341] [Ideas]
For people like me that mostly need SFTP repos, it's cool to be able to configure the SSH config in one file, so we just need to set the `sftp:host:target` repo URI at creation.

Example

- Create a `ssh` directory

```
mkdir ./ssh
```

- Create the `ssh/config` file

```
Host host
    Hostname truc.chouette.fr
    Port 12345
    User me
```

- Generate the `known_hosts`

```
ssh-keyscan -H -p 12345 truc.chouette.fr > ./ssh/known_hosts
```

- Create a key

```
ssh-keygen -f ./ssh/id_rsa
```

- Set the expected permissions

```
chmod 700 ./ssh
chmod 600 ./ssh/*
sudo chown -R root:root ./ssh/
```

- Mount the directory at the default location

**compose.yml**
```yml
services:
  backrest:
#...
    volumes:
#...
      - ./ssh:/root/.ssh
```

- Now we can create a repo with just `sftp:host:target` as URI

Maybe this would deserve a README section, what do you think ?

**Comment by garethgeorge** [comment]
Definitely worth a README section or perhaps a cookbook at https://garethgeorge.github.io/backrest/cookbooks/ 

Writing up more docs is solidly on my todo list -- any contributions here are also very welcome! SSH configuration is perhaps one of the most asked about issues and definitely deserves some working examples.

**Reply by Glideh** [reply]
I can do I PR if needed

**Reply by garethgeorge** [reply]
Went ahead and wrote this up in https://garethgeorge.github.io/backrest/cookbooks/ssh-remote , thanks again for contributing the initial tips / guide of what worked for you.

**Reply by Stephane-30** [reply]
Hi @garethgeorge . Thank you for the tutorial. It was exactly what I was looking for to be able to backup some directories from my little server to my NAS via sftp. However, it took me two hours to work out that I have to change ownership of ssh dir to root:root. I finally found the hint in this tread. Maybe you want to add that to your tutorial under point 5. :-)

**Comment by raphpa** [comment]
This can also be done with the restic additional sftp args like:

`-o sftp.args="-o UserKnownHostsFile=/config/known_hosts -i /config/id_ed25519"`

Also the port is not recognized if you do not use double slashes after `sftp:` (as recommended in the documentation). For Hetzner storage boxes for example it would be `sftp://u123456-sub123@u123456.your-storagebox.de:23//home`

**Reply by garethgeorge** [reply]
Thanks for calling this out -- it's clear that there's a lot of inconsistent documentation re: backrest and sftp setup, it's a major painpoint. I've filed https://github.com/garethgeorge/backrest/issues/620 to track writing up a guide
